src/main/java/com/mojang/datafixers/functions/PointFree.java

```diff
@@ -16,7 +16,7 @@
     private Function<DynamicOps<?>, T> value;
 
     @SuppressWarnings("ConstantConditions")
-    public Function<DynamicOps<?>, T> evalCached() {
+    public synchronized Function<DynamicOps<?>, T> evalCached() {
```

Comment:
This seems too heavy-weight, since evalCached should be a hot spot. There might be a better option, like https://en.wikipedia.org/wiki/Double-checked_locking

Reply:
This is something I was concerned about too and was considering a double checked lock also. I will update it.

Wasn't able to detect any performance concerns considering we have usage of datafixers all done on the thread pool now, but helping improve conversion speed is always good.